internal/storage/oauth_client.go

```diff
@@ -0,0 +1,162 @@
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/ory/fosite"
+	"go.infratographer.com/identity-api/internal/types"
+)
+
+var oauthClientCols = struct {
+	ID       string
+	TenantID string
+	Name     string
+	Secret   string
+	Audience string
+}{
+	ID:       "id",
+	TenantID: "tenant_id",
+	Name:     "name",
+	Secret:   "secret",
+	Audience: "audience",
+}
+
+var (
+	oauthClientColumns = []string{
+		oauthClientCols.TenantID,
+		oauthClientCols.Name,
+		oauthClientCols.Secret,
+		oauthClientCols.Audience,
+	}
+	oauthClientColumnsStr = strings.Join(oauthClientColumns, ", ")
+)
+
+type oauthClientManager struct {
+	db     *sql.DB
+	hasher fosite.Hasher
+}
+
+// ClientAssertionJWTValid implements fosite.ClientManager
+func (*oauthClientManager) ClientAssertionJWTValid(ctx context.Context, jti string) error {
+	panic("unimplemented")
+}
+
+// GetClient implements fosite.ClientManager
+func (s *oauthClientManager) GetClient(ctx context.Context, id string) (fosite.Client, error) {
+	return s.LookupOAuthClientByID(ctx, id)
+}
+
+// SetClientAssertionJWT implements fosite.ClientManager
+func (*oauthClientManager) SetClientAssertionJWT(ctx context.Context, jti string, exp time.Time) error {
+	panic("unimplemented")
+}
+
+func newOAuthClientManager(config Config, db *sql.DB) (*oauthClientManager, error) {
```

Comment:
`config` isn't used in this function, and #94 removes references to it entirely for other storage engine things; we can probably get rid of it here.